module/plugins/hoster/FileAl.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+#
+# ATTENTION: if you cannot see the interactive captcha (on firefox), make sure to activate/install X-Frame-Options Header:
+# https://addons.mozilla.org/en-US/firefox/addon/ignore-x-frame-options-header/
+
+from module.plugins.internal.XFSHoster import XFSHoster
+import re
+
+
+class FileAl(XFSHoster):
+    __name__    = "FileAl"
+    __type__    = "hoster"
+    __version__ = "0.01"
+    __status__  = "testing"
+    
+    __pattern__ = r'https?://(?:www\.)?file\.al/\w{12}'
+    
+    __description__ = """File.al hoster plugin"""
+    __license__     = "GPLv3"
+    __authors__     = [("igel", "")]
+
+    PLUGIN_DOMAIN = "file.al"
+    LINK_PATTERN     = r'direct link.*?<a [^>]*href="(.+?)".*?>Click here to download', re.MULTILINE | re.DOTALL
+    WAIT_PATTERN     = r'countdown.*?seconds.*?(\d+)'
+    # "extend" the XFSHoster dict
+    RECAPTCHA_PATTERN = r"g-recaptcha.*?sitekey=[\"']([^\"]*)"
+    PREMIUM_ONLY_PATTERN  = r'(?:[Pp]remium Users only|can download files up to.*only)'
+    LOCAL_LOCATION = r'check_ip'
```

Comment:
What is LOCAL_LOCATION?

Reply:
Ah, you're right, I forgot I included that in `XFSHoster`: sometimes, fileal would answer with a `location=` in the header while still advancing in the `op`-chain (I don't remember when exactly or why it did that). With the standard `XFSHoster`, this leads to pyload trying to download the HTML-file given in the `location=` header field, which is not what we want. Since, for FileAl, the `location=...` header field would contain the substring `check_ip` somewhere, I decided to check for this substring and, if found, to not use it as direct link but to follow it in order to advance in the `op`-chain. I figured this may happen for other XFSHoster as well, so I included the code for that in `XFSHoster.py`.

This having said, I'm not sure whether it is still ncessary or not. Do you want me to re-submit without `LOCAL_LOCATION` (seeing that changes to `XFSHoster.py` are probably not allowed at this point)?